aws/network_acl_entry.go

```diff
@@ -43,7 +43,7 @@ func expandNetworkAclEntries(configured []interface{}, entryType string) ([]*ec2
 		}
 
 		// Specify additional required fields for ICMP
-		if p == 1 {
+		if p == 1 || p == 58 {
```

Comment:
IMO this wants a comment/constant explaining what 1 and 58 represent

Reply:
I created a technical debt issue for this: #6302